src/linter/root.go

```diff
@@ -63,7 +63,8 @@ type RootWalker struct {
 
 	currentClassNode node.Node
 
-	disabledFlag bool // user-defined flag that file should not be linted
+	isDisabled   bool            // user-defined flag that file should not be linted
```

Comment:
This should probably be called `allowDisabled` as it's not disabled if there were not `linter disable` comments inside a file. Also, I would store a regexp here as you pass it into parse contents function anyway. Then we can try to match a regexp only if we encounter the `@linter disable` comment (which is rare) instead of matching every filename against it.